packages/auto-spies-core/src/create-function-auto-spy.ts

```diff
@@ -47,11 +48,13 @@ export function createFunctionAutoSpy<ReturnType, LibSpecificType>(
   const calledWithObject: CalledWithObject = {
     wasConfigured: false,
     argsToValuesMap: new Map(),
+    functionSpyName: name,
   };
 
   const mustBeCalledWithObject: CalledWithObject = {
     wasConfigured: false,
     argsToValuesMap: new Map(),
+    functionSpyName: name,
```

Comment:
is it really required as part of the object?

can't we just add another parameter to the `returnTheCorrectFakeValue` and pass the `name` (maybe we'll rename `name` to `functionName` to make it clearer)